producer/buffer/buffer.go

```diff
@@ -171,24 +176,41 @@ func (b *buffer) Init() {
 }
 
 func (b *buffer) cleanupUntilClose() {
-	ticker := time.NewTicker(b.opts.CleanupInterval())
+	ticker := time.NewTicker(
```

Comment:
Kinda strange how this is set up, do you intend to try N times every N seconds? Why is that better than a normal retrier?

Reply:
With a retrier + ticker here, it will by default perform cleanup at the tick interval, but when no progress can be made, it will start to backoff and eventually cleanup once every `maxBackoff`. Once it can progress again, the cleanup will be performed every `initialBackoff` again.

Reply:
I see, so if that happens (i.e., cleaning up every `maxBackoff` and then coming back down `initialBackoff`), would the ticker have fired multiple times by then causing you to immediately clean up several times afterwards?

Reply:
In that case, the ticker will have one tick buffered, so there will be one immediate cleanup rather than several